### DISCUSSION TITLE
src_ep = dst_ep for _send_request_raw()

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy-znp#194] [Q&A]
I'm trying to understand why `src_ep` must be equal to `dst_ep` in zigpy_znp ?

https://github.com/zigpy/zigpy-znp/blob/e6e2c42981c8a04303db6d28ef503a6b0778d667/zigpy_znp/zigbee/application.py#L811

When _send_request_raw() is called, src_ep and dst_ep are set by the above layer, is there any particular restriction from ZNP to force that ?

**Comment by puddly** [comment]
This is due to a problem still in zigpy:

https://github.com/zigpy/zigpy/blob/8a2c7548065367d7e94ddf5857652ca1cd087e70/zigpy/endpoint.py#L240-L241

The packet API rewrite didn't include everything.

**Comment by pipiche38** [comment]
Is there anything we can help you here ?
